controlplane/eks/api/v1beta2/awsmanagedcontrolplane_webhook.go

```diff
@@ -392,6 +394,31 @@ func (r *AWSManagedControlPlane) validateDisableVPCCNI() field.ErrorList {
 	return allErrs
 }
 
+func (r *AWSManagedControlPlane) validateFilterPrivateSubnets() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if r.Spec.FilterPrivateSubnets {
+		boolField := field.NewPath("spec", "filterPrivateSubnets")
+		allPublic := true
+		if r.Spec.NetworkSpec.Subnets != nil {
+			for _, subnet := range r.Spec.NetworkSpec.Subnets {
+				if !subnet.IsPublic {
```

Comment:
I should be able to use pre defined `subnets.FilterPublic() /. subnets.FilterPrivate()` methods here.